### PR TITLE
feature: verbose violations

### LINKED ISSUE #15: Support ability to propogate stacktraces from NoThrow transforms
> it would be good to propagate the stacktrace in some manner for NoThrow transforms (not by throwing it, but by capturing it and outputting it somewhere the user could log it if needed). We don't really want to debug in prod but if we do see an error that is hard to repro locally, getting a stacktrace in the logs can be invaluable. (@ericphanson)

### PR DESCRIPTION
**Pull request by ericphanson**
Allows passing `verbose_violations=true` to ask TS to generate much more detailed violation strings. 

* This is configurable because they include full stacktraces, so can be hundreds of lines long, and aren't appropriate for doctests.
* This defaults to `false`, since I'm not sure if folks are testing the violations they get exactly and don't want to break those tests. Also maybe it is too verbose for some folks.
    * Also want to go with the least-controversial option to get this in asap

closes #15

### CODE REVIEW

File: src/nothrow.jl [hunk 1]
```diff
@@ -248,28 +248,38 @@ will fail (i.e., return a `NoThrowResult{Missing}` if:
...
         return NoThrowResult(; violations)
     end
 
     # Do transformation
     result = try
         NoThrowResult(ntt.transform_spec.transform_fn(input))
-    catch e
-        return NoThrowResult(; violations="Unexpected violation. Details: $e")
+    catch
+        stack = current_exceptions()
```
Comment (franzfurbass) [inline src/nothrow.jl:280]: I wonder if this also should be switched with `verbose_violations = true` and `string(e)` used otherwise.

Comment (franzfurbass) [reply src/nothrow.jl:280]: it is a `NoThrowResult` call which suggests there is no exception I guess.

**Review by franzfurbass — COMMENTED** [review]
I think I am fine with it because the default behaviour is the same.
- except this part there that seems to change for both cases: https://github.com/beacon-biosignals/TransformSpecifications.jl/pull/31/files#r1352789979

File: src/nothrow.jl (hunk 1 above)
Comment (ericphanson) [reply src/nothrow.jl:280]: oops, I meant to check this one too

Comment (ericphanson) [reply src/nothrow.jl]: good catch! pushed a fix

**Pull request merged**